python/histone_utility.py

```diff
@@ -0,0 +1,355 @@
+        histone_list: list of histones which correlation study will be performed
+        gene_data: dataframe containing genes which you would like to compare
+        histone_data: dataframe containing histones which you would like to compare
+        dtype: 'numpy' or 'df', numpy returns a matrix, and df returns a dataframe
+            
+    OUTPUT
+    
+        Rmat: R-value matrix
+        Pmat: p-value matrix
+        Rdf: R-value dataframe
+        Pdf: p-value dataframe
+        
+    """
+    Rmat = (len(gene_list), len(histone_list))
+    Rmat = np.zeros(Rmat)
+    Pmat = Rmat[:]
+    i = 0 
+    
+    for gene in gene_list:
+        j = 0
+        for histone in histone_list:
+            gene_data.loc[gene] = gene_data.loc[gene].fillna(method='ffill')
+            histone_data.loc[histone] = histone_data.loc[histone].fillna(method='ffill')
+            correlation = pearsonr(gene_data.loc[gene], histone_data.loc[histone])
+            Rmat[i][j] = correlation[0]
+            Pmat[i][j]
+            j = j+1
+        i = i+1
+        
+    if dtype=='numpy':
+        return Rmat, Pmat
+    
+    if dtype=="df"  :  
+        Rdf = pd.DataFrame(Rmat)
+        Rdf.columns = histone_list
+        Rdf.insert(0, 'Genes', gene_list, True)
+        Rdf = Rdf.set_index('Genes')
+    
+        Pdf = pd.DataFrame(Pmat)
+        Pdf.columns = histone_list
+        Pdf.insert(0, 'Genes', gene_list, True)
+        Pdf = Pdf.set_index('Genes')
+        
+        return Rdf, Pdf
+    
+    
+def tissue_analysis(cell_line_dict, common_cellline, histone_list, gene_list, 
+                    data1, data2, tissue_type ):
+    """ This function is only necessary if you would like to perform some basic 
+    tissue analysis on data. The dictionary input should be a dictionary which 
+    matches cell lines to their specific tissue. 
+    
+    Common_cellline should be a list of the common_celllines 
+    between the two data sets. 
+    
+    Both data inputs should be dataframes created from the pearson_dfs
+    function and the tissue_type will correspond to the type of tissue wishing 
+    to be analyzed. The output is a dataframe which can be graphed.
+    
+    This may not give you the best idea of tissue analysis as most datasets 
+    vary in the amount of celllines corresponding to each tissue. 
+    
+    
+    INPUT
+        cell_line_dict: dictionary of cell lines and corresponding tissue
+        common_cellline: list of cell lines both data sets have in common
+        histone_list: list of histone markers wishing to be used
+        gene_list: list of genes wishing to be used
+        data1: the first data set being compared
+        data2: the second data set being compared
+        tissue_type: type of tissue wishing to isolate
+        
+    OUTPUT:
+        matrix: new matrix with only cell lines corresponding to tissue_type
+        
+    """
+    
+    df1 = data1.copy()
+    df2 = data2.copy()
+    for celllines in common_cellline:
```

Comment:
Can be vectorized using iloc

Reply:
I think I need another hint on this one cause I cant figure out the logic like I understand what youre saying but like im confused on how I would do it

Reply:
look up "deleted columns based on list pandas" on google. there are many ways to tackle this problem.